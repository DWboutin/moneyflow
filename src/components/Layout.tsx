import React, { ReactNode } from "react";
import styled, { ThemeProvider } from "styled-components";
import { ToastContainer } from "react-toastify";

import "normalize.css";
import "boxicons/css/boxicons.css";
import "react-datepicker/dist/react-datepicker.css";
import "react-toastify/dist/ReactToastify.css";

import GlobalStyle from "../styles/globalStyle";
import { theme } from "../styles/styles";
import Nav from "./Nav/Nav";

interface ContainerProps {}

const Container = styled.div<ContainerProps>`
  display: flex;
  flex: 1;
  min-height: 100vh;
`;

const Content = styled.div<ContainerProps>`
  display: flex;
  flex: 1;
  padding: 50px 80px;
`;

interface Props {
  children: ReactNode;
}

const Layout: React.FunctionComponent<Props> = ({ children }) => {
  return (
    <ThemeProvider theme={theme}>
      <Container>
        <GlobalStyle />
        <Nav />
        <Content>{children}</Content>
      </Container>
      <ToastContainer />
    </ThemeProvider>
  );
};

export default Layout;
