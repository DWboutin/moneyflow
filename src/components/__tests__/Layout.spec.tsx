import React from "react";
import { render, RenderResult } from "@testing-library/react";

import Layout, { Props } from "../Layout";

describe("<Layout />", () => {
  let renderResult: RenderResult;

  const renderLayout = ({ children }: Props) => {
    renderResult = render(<Layout>{children}</Layout>);
  };

  describe("default rendering", () => {
    beforeEach(() => {
      renderLayout({ children: "Content" });
    });

    it("should match snapshot", () => {
      expect(renderResult.container).toMatchSnapshot();
    });
  });
});
