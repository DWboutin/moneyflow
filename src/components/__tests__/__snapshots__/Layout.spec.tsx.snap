// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> default rendering should match snapshot 1`] = `
<div>
  <div
    class="sc-furwcr hbJNlB"
  >
    <div
      class="sc-gsDKAQ kGCKzU"
    >
      <div>
        <div
          class="sc-dkPtRN dTUoTS"
        >
          MoneyFlow
        </div>
        <div
          class="sc-hKwDye cFKaep"
        >
          <div
            class="sc-eCImPb dZgOzE"
          />
          <div
            class="sc-jRQBWg ovfEn"
          >
            Welcome back
          </div>
          <div
            class="sc-gKclnd eTIzFW"
          >
            Michael Gregson
          </div>
        </div>
      </div>
      <div
        class="sc-iCfMLu drCLaC"
      >
        <div>
          <a
            class="sc-bdvvtL hEsoLv"
            href="/"
          >
            <i
              class="bx bxs-dashboard bx-sm"
            />
            <span>
              Dashboard
            </span>
          </a>
          <a
            class="sc-bdvvtL hEsoLv"
            href="/record-payment"
          >
            <i
              class="bx bx-message-square-add bx-sm"
            />
            <span>
              Record Payment
            </span>
          </a>
          <a
            class="sc-bdvvtL hEsoLv"
            href="/tracking-expenses"
          >
            <i
              class="bx bx-notepad bx-sm"
            />
            <span>
              Tracking Expenses
            </span>
          </a>
        </div>
        <a
          class="sc-bdvvtL hEsoLv"
          href="/log-out"
        >
          <i
            class="bx bx-log-out bx-sm"
          />
          <span>
            Log out
          </span>
        </a>
      </div>
    </div>
    <div
      class="sc-pVTFL iAoSIl"
    >
      Content
    </div>
  </div>
  <div
    class="Toastify"
  />
</div>
`;
